### ISSUE TITLE
Could not export notes: Cannot read property 'type_' of undefined

### ISSUE THREAD

**Issue by ZSZ2000** [laurent22/joplin#2513]
## Environment

Joplin version:1.0.179
Platform:Windows
OS specifcs:1903

## Steps To Reproduce

1. Sorry, I don't know how to reproduce this problem
2. When I write notes, some of my notes will disappear after a save synchronization. These notes can be searched, but they are not displayed in the notebook.
3. And the note can't be export, prompt "Could not export notes: Cannot read property 'type_' of undefined"

## Logfile

[log.txt](https://github.com/laurent22/joplin/files/4209640/log.txt)
[log-autoupdater.txt](https://github.com/laurent22/joplin/files/4209641/log-autoupdater.txt)
[log-clipper.txt](https://github.com/laurent22/joplin/files/4209642/log-clipper.txt)
[log-database.txt](https://github.com/laurent22/joplin/files/4209643/log-database.txt)

**Comment by mic704b** [comment]
What is a "save synchronization"?

Is that when you selected "Synchronisation" in the File menu (or Ctrl-S)?

**Comment by ZSZ2000** [comment]
Yes, it is. My notes can't be displayed in my notebook after the synchronization.

**Comment by mic704b** [comment]
There may be two isues here.
1. Why did your sync fail and did it remove your notes?
2. If your notes are gone, how did they appear in search results?

If your notes are gone, that would explain why you can't export them.

Which sync target do you use?

**Comment by ZSZ2000** [comment]
Webdev
https://dav.jianguoyun.com/dav/

**Comment by ZSZ2000** [comment]
PS:
1. I don't think the notes was deleted. Probably the notebooks' tree was broken, so it couldn't be displayed but can be searched.
2. For similar reasons, these notes can't display some pictures even when they are displayed after searching, but I checked the corresponding files, which exist. When these notes was exported, the error would occurs.
3. I think there may be a problem with the database, it seems that there was a conflict during synchronization, but the conflict folder only shows one note. Most of the hidden notes are not shown. I tried again, these hidden notes will be displayed in the label folders, can also be searched, but can't be displayed in the tree folders.
